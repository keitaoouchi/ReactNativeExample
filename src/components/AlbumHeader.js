import React from "react";
import {Animated, Easing, Text, TouchableOpacity, View} from "react-native";
import Util from "../Util";
import ListenOnSpotifyButton from "./ListenOnSpotifyButton";

export default class AlbumHeader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fadeAnim: new Animated.Value(0),
    };
  }

  render() {
    const album = this.props.album;
    const albumDetail = this.props.albumDetail;
    const onPressArtistName = this.props.onPressArtistName;
    const spotifyURL = !!albumDetail && albumDetail.spotifyURL || '';
    return (
      <View style={ style.root }>
        <Animated.Image
          source={ {uri: album.coverImageURL}}
          style={ [style.image, {opacity: this.state.fadeAnim}]}
          onLoadEnd={() => {
            Animated.timing(
              this.state.fadeAnim,
              {
                toValue: 1,
                easing: Easing.easeInOut,
              },
            ).start();
          }}
        />
        <Text style={ style.albumTitle }>{album.title}</Text>
        <View style={ style.artistName }>
          <Text style={ style.by }>By</Text>
          <TouchableOpacity
            style={ style.button }
            onPress={onPressArtistName}>
            <View>
              <Text style={ style.buttonText }>{album.artist.name}</Text>
            </View>
          </TouchableOpacity>
        </View>
        <Text style={ style.miscInfo }>{Util.releasedYear(albumDetail)}</Text>
        <ListenOnSpotifyButton url={spotifyURL}/>
      </View>
    );
  }
}

const style = {
  root: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  by: {opacity: 0.5, fontSize: 16, color: 'white'},
  button: {padding: 8},
  buttonText: {color: 'white', fontWeight: 'normal', fontSize: 16},
  image: {width: 240, height: 240, margin: 20},
  albumTitle: {fontSize: 24, margin: 12, color: 'white', fontWeight: 'bold'},
  artistName: {
    margin: 8,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  miscInfo: {fontSize: 12, margin: 8, opacity: 0.5, color: 'white'},
};