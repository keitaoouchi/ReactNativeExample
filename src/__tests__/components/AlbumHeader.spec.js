import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import AlbumHeader from "../../../src/components/AlbumHeader";
import AlbumFactory from "../factories/AlbumFactory";
import AlbumDetailFactory from "../factories/AlbumDetailFactory";

describe('AlbumHeader', () => {
  it('renders correctly', () => {
    const album = AlbumFactory.build('Album');
    const albumDetail = AlbumDetailFactory.build('AlbumDetail');
    const tree = renderer.create(
      <AlbumHeader album={album} albumDetail={albumDetail}
                   onPressArtistName={() => {
                   }}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
