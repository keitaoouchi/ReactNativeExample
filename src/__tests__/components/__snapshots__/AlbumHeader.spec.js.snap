// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "https://example.com/1",
      }
    }
    style={
      Object {
        "height": 240,
        "margin": 20,
        "opacity": 0,
        "width": 240,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
        "fontWeight": "bold",
        "margin": 12,
      }
    }
  >
    AlbumTitle-1
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 8,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "opacity": 0.5,
        }
      }
    >
      By
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "padding": 8,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "normal",
            }
          }
        >
          Name-1
        </Text>
      </View>
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 12,
        "margin": 8,
        "opacity": 0.5,
      }
    }
  >
    2017年
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1ed760",
            "borderRadius": 22,
            "height": 44,
            "margin": 12,
            "padding": 8,
            "width": 200,
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "margin": 4,
            "textAlign": "center",
          }
        }
      >
        Spotifyで聴く
      </Text>
    </View>
  </View>
</View>
`;
